src/software/world/ball.cpp

```diff
@@ -30,14 +35,19 @@ TimestampedBallState Ball::currentState() const
     return states_.front();
 }
 
+const std::shared_ptr<BallModel> &Ball::getBallModel() const
+{
+    return ball_model_;
+}
+
 void Ball::updateState(const TimestampedBallState &new_state)
 {
     if (!states_.empty() && new_state.timestamp() < lastUpdateTimestamp())
     {
         throw std::invalid_argument(
             "Error: Trying to update ball state using a state older then the current state");
     }
-
+    ball_model_ = std::make_shared<LinearBallModel>(LinearBallModel(new_state.state()));
```

Comment:
do we construct this every-time to avoid having an `updateModel` function be part of the interface?

Reply:
the goal is that this function goes away in the world rearch